src/main/java/se/sundsvall/precheck/service/utils/PreCheckUtil.java

```diff
@@ -0,0 +1,146 @@
+package se.sundsvall.precheck.service.utils;
+
+import generated.client.citizen.CitizenAddress;
+import generated.client.citizen.CitizenExtended;
+import generated.client.partyAssets.Asset;
+import jakarta.validation.constraints.NotNull;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.zalando.problem.Problem;
+import se.sundsvall.precheck.api.model.PreCheckResponse;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.zalando.problem.Status.BAD_REQUEST;
+import static se.sundsvall.precheck.constant.Constants.ASSET_TYPE_EXISTS_ERROR_MESSAGE;
+import static se.sundsvall.precheck.constant.Constants.CORRECT_ADDRESS_TYPE;
+import static se.sundsvall.precheck.constant.Constants.NO_VALID_MUNICIPALITY_ID_FOUND;
+
+public class PreCheckUtil {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PreCheckUtil.class);
+
+    private PreCheckUtil() {
+        throw new IllegalStateException("Utility class");
+    }
+    public static boolean checkResourceAvailability(ResponseEntity<CitizenExtended> citizen, ResponseEntity<List<Asset>> party) {
+        if (citizen == null || party == null) {
+            LOGGER.error("Citizen or Party is null during resource availability check");
+            return true;
+        }
+
+        try {
+            LOGGER.info("Input data checkResourceAvailability: Citizen: {}    Party: {}", citizen.getStatusCode(), party.getStatusCode());
+
+            boolean isCitizenBodyPresent = citizen.getBody() != null && !citizen.getBody().equals(new CitizenExtended());
+            boolean areStatusCodesAcceptable = citizen.getStatusCode() == HttpStatus.OK && party.getStatusCode() == HttpStatus.OK;
+
+            if (isCitizenBodyPresent && areStatusCodesAcceptable) {
+                LOGGER.info("Both citizenBody and partyBody are valid.");
+                return false;
+            }
+            LOGGER.info("Either citizenBody or partyBody is invalid.");
+
+            return true;
```

Comment:
Move this to row 54 and remove the "return true" on row 52 as well.

Reply:
That is a way better way to do it.

Thanks for the help 🥇